common/lib/opentelemetry/common/utilities.rb

```diff
@@ -69,7 +69,7 @@ def utf8_encode(string, binary: false, placeholder: STRING_PLACEHOLDER)
       #
       # @return [String]
       def truncate(string, size)
-        string.size > size ? "#{string[0...size - 3]}..." : string
+        string.is_a?(String) && string.size > size ? "#{string[0...size - 3]}..." : string
```

Comment:
I see that there is a type check in the case statement where this function is used. 

Is truncate being used in other places where it does not receive a String? If not then I do not think this needs to change.

Reply:
It is called elsewhere, yes, including in instrumentation libraries. So I figured it was better to not require the check be added everywhere. The reason it is also checked in that other case statement is only because there is an additional step to do if the value is an array, I figured if it was already checking if the value was an array there it may as well not even call the method if it wasn't a string.

Reply:
This kind of defensive programming is not typical in Ruby. The method docs state that the first argument is a `String`. Callers should ensure that they only call it with a `String`. We shouldn't be type-checking that argument here.

If another caller is calling it with invalid arguments, that should be considered a bug in the caller, and we should fix the caller.

Reply:
Ok, I can change it, but that is also true in Erlang and we still do checks like this more so than normal because as a library for telemetry we want to be as silent and unseen as possible. So I thought it might be similar in otel ruby.